src/components/SwapForm/index.tsx

```diff
@@ -301,7 +305,10 @@ export default function Swap() {
     }
     // Do evaluate usd select tokens prices
     if (data.size && !isLoading) {
-      data.forEach((token) => {
+      const getAssetList: TokenListWithNotSelectableToken[] = []
+      data.forEach((value) => getAssetList.push(value))
+      const tieNativeToWrapAssetList = tieNativeToWrapToken(getAssetList)
```

Comment:
Why not just, pass data as an arg to tieNativeToWrapToken. It creates a new array and doesn't seem to mutate an incoming one. As for typing you can just assert it if that is the problem.